src/daft-core/src/utils/identity_hash_set.rs

```diff
@@ -0,0 +1,23 @@
+use std::hash::{BuildHasherDefault, Hasher};
+
+pub type IdentityBuildHasher = BuildHasherDefault<IdentityHasher>;
+
+#[derive(Default)]
+pub struct IdentityHasher {
```

Comment:
Added docstrings for the purpose of this struct.

I think keeping the name as `IdentityHasher` should suffice since this can be used to hash things of multiple different types. For example, you can construct a `HashMap<String, IdentityHasher>` if you wish.

The primary concern here is notifying to the end-user that what you are inserting into this map *will not be hashed again*. I think my docstring has sufficiently pushed that point across.

Let me know what you think!